lib/backoff/decorr.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backoff
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// Decorr is a "decorrelated jitter" inspired by https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/.
+type decorr struct {
```

Comment:
You might want to use something like the [Clockwork](https://github.com/jonboulle/clockwork) package to allow a mock clock to be injected for testing

Reply:
Done! Clockwork was added.